src/features/home/components/Map/Map.tsx

```diff
@@ -0,0 +1,144 @@
+import classNames from 'classnames/bind';
+import { GeoJSONSource } from 'mapbox-gl';
+import { useRef, useState, useEffect } from 'react';
+import ReactMapGL, { Layer, MapRef, Source, useMap } from 'react-map-gl';
+
+import 'mapbox-gl/dist/mapbox-gl.css';
+import { Experience } from '@/shared/types/Experience';
+
+import Slider from '../Slider';
+
+import {
+  clusterLayer,
+  unclusteredPointLayer,
+  clusterCountBackgroundLayer,
+  clusterCountLayer,
+} from './layers';
+import styles from './Map.module.scss';
+
+const cx = classNames.bind(styles);
+
+const mapboxAccessToken = process.env.NEXT_PUBLIC_MAPBOX_TOKEN?.replace(
+  / /g,
+  ''
+);
+
+type MapProps = {
+  records: Experience[];
+};
+
+const MapImage = ({ id, src }: { id: string; src: string }) => {
+  const { current: map } = useMap();
+
+  if (map && !map.hasImage(id)) {
+    map.loadImage(src, (error, image) => {
+      if (error) throw error;
+      if (!map.hasImage(id) && image) {
+        map.addImage(id, image);
+      }
+    });
+  }
+
+  return null;
+};
+
+const Map = ({ records }: MapProps) => {
+  const mapRef = useRef<MapRef>(null);
+  const [selectedItems, setSelectedItems] = useState<Experience[]>([]);
+  const [data, setData] = useState({
+    type: 'FeatureCollection',
+    features: records?.map((record: Experience) => ({
+      type: 'Feature',
+      properties: record,
+      geometry: {
+        type: 'Point',
+        coordinates: [Number(record.lat), Number(record.lng)],
+      },
+    })),
+  });
```

Comment:
record가 prop이기 때문에 변경되는 경우 컴포넌트가 렌더링 되기 때문에 data라는 상태를 따로 선언해주지 않아도 되지않을까요?? (아래 useEffect부분도 제거해도 될것 같아요
```suggestion
  const data = useMemo(
    () => ({
      type: 'FeatureCollection',
      features: records?.map((record: Experience) => ({
        type: 'Feature',
        properties: record,
        geometry: {
          type: 'Point',
          coordinates: [Number(record.lat), Number(record.lng)],
        },
      })),
    }),
    [records]
  );
```

Reply:
엇 맞네요 반영하겠습니다